### PR TITLE
Update doc for CA1010 to mention new configurability

### LINKED ISSUE #4989: CA1010 - New configurability section
With the release of fxcop 3.x, a new user-option is provided to allow providing additional set of interfaces which requires the generic pendant to be implemented (https://github.com/dotnet/roslyn-analyzers/pull/3258)

See https://github.com/dotnet/roslyn-analyzers/pull/3258

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 6f4f044f-87ac-75f0-99e1-ef14bcbdc532
* Version Independent ID: 5c6766a2-8cb3-695e-968a-ae9a549aa82b
* Content: [CA1010: Collections should implement generic interface - Visual Studio](https://docs.microsoft.com/en-us/visualstudio/code-quality/ca1010?view=vs-2019)
* Content Source: [docs/code-quality/ca1010.md](https://github.com/MicrosoftDocs/visualstudio-docs/blob/master/docs/code-quality/ca1010.md)
* Product: **visual-studio-windows**
* Technology: **vs-ide-code-analysis**
* GitHub Login: @Mikejo5000
* Microsoft Alias: **mikejo**

### PR DESCRIPTION
**Pull request by Evangelink**
Fix #4989

### CODE REVIEW

**Comment by PRMerger16** [comment]
@Evangelink : Thanks for your contribution! The author(s) have been notified to review your proposed change.

**Comment by ktoliver** [comment]
Hi @Mikejo5000 Note that the PR review team doesn't have a way to hold PRs, if the set submitted today is approved, until they are ready to be published. If you accept the updates, the content might be best off moved to the private repo where you can add sign-off when the content is ready to go live. Thanks.

**Comment by Mikejo5000** [comment]
@mikadumont @mavasani Are these updates (this and related PRs for FxCop 3.x) ready to go live? Looks like FxCop 3.x is still in beta. Generally, we wouldn't publish prior to GA. Let me know what you think. Thanks!

**Comment by mavasani** [comment]
@Mikejo5000 - yes, we should wait till a public 3.0.x is released before merging these PRs. Thanks!

**Comment by Mikejo5000** [comment]
Any ETA for 3.0.x? We will have folks asking about it if the PRs are on the public site for very long.

**Comment by mavasani** [comment]
We just released a beta this week, and will be waiting for couple of weeks before releasing the stable 3.0.0 packages.

File: docs/code-quality/ca1010.md [hunk 1]
```diff
@@ -60,6 +60,21 @@ dotnet_code_quality.ca1010.api_surface = private, internal
 
 You can configure this option for just this rule, for all rules, or for all rules in this category (Design). For more information, see [Configure FxCop analyzers](configure-fxcop-analyzers.md).
 
+### Additional required generic interfaces
+
+You can configure the List (separated by '|') of interface names with their required generic fully qualified interface (separated by '->').
```
Comment (mavasani) [inline docs/code-quality/ca1010.md:65]: Shouldn't this be:
```suggestion
You can configure the List of interface names (separated by '|') with their required generic fully qualified interface (separated by '->').
```

I'll let the doc experts confirm the correct sentence formation.

File: docs/code-quality/ca1010.md [hunk 2]
```diff
@@ -28,7 +28,7 @@ ms.workload:
 
 A type implements the <xref:System.Collections.IEnumerable?displayProperty=fullName> interface but does not implement the <xref:System.Collections.Generic.IEnumerable%601?displayProperty=fullName> interface, and the containing assembly targets .NET. This rule ignores types that implement <xref:System.Collections.IDictionary?displayProperty=fullName>.
 
-By default, this rule only looks at externally visible types, but this is [configurable](#configurability).
+By default, this rule only looks at externally visible types, but this is [configurable](#configurability). You can also configure additional interfaces requiring there generic pendent to be implemented.
```
Comment (mavasani) [inline docs/code-quality/ca1010.md:31]: I think this needs grammatical fixes.

Comment (Evangelink) [reply docs/code-quality/ca1010.md:31]: @ktoliver Could you please suggest a better wording so that I can fix this part?

Comment (ktoliver) [reply docs/code-quality/ca1010.md:31]: "By default, this rule looks only at externally visible types, but this is configurable. You also can configure additional interfaces, which requires the interface generic pendent to be implemented."

?

Comment (mavasani) [reply docs/code-quality/ca1010.md:31]: What is `generic pendent`? I am not sure I understand...

Comment (Evangelink) [reply docs/code-quality/ca1010.md:31]: Google translate isn't my friend... I meant something like the `generic match`: `IEnumerable` -> `IEnumerable<T>`.

File: docs/code-quality/ca1010.md (hunk 1 above)
Comment (Mikejo5000) [reply docs/code-quality/ca1010.md:65]: Yes, I'll commit that.

File: docs/code-quality/ca1010.md (hunk 2 above)
Comment (Mikejo5000) [reply docs/code-quality/ca1010.md:31]: Probably I'd say "generic equivalent".

Comment (Mikejo5000) [inline docs/code-quality/ca1010.md:31]:
```suggestion
By default, this rule only looks at externally visible types, but this is [configurable](#configurability). You can also configure additional interfaces requiring their generic equivalent to be implemented.
```

Comment (Mikejo5000) [reply docs/code-quality/ca1010.md:31]: After reading the new content, trying to clarify phrasing a bit.

Comment (Mikejo5000) [reply docs/code-quality/ca1010.md:31]: Let me know if it seems accurate. Not sure that it covers this example: ISomething->System.Collections.Generic.IEnumerable`\``1,
which doesn't seem like the generic equivalent.
Maybe this is better and simpler:
"You can also configure additional interfaces to require that a generic interface be implemented."

Comment (mavasani) [reply docs/code-quality/ca1010.md:31]: > "You can also configure additional interfaces to require that a generic interface be implemented."

LGTM

**Comment by PRMerger10** [comment]
@Mikejo5000 : Thanks for your contribution! The author(s) have been notified to review your proposed change.

**Comment by Mikejo5000** [comment]
#sign-off

**Pull request merged**